### PR TITLE
v3 of the editor

### PR DESCRIPTION
**Pull request by burtonator**
This also duplicates NewThreadForm into NewThreadFormLegacy and NewThreadFormModern so you don't really need to fully audit those two as NewThreadFormModern is the main one and most of the code didn't change.

Also, everything here should be safely behind a feature toggle so nothing is being turned on yet.

I have 1-2 more revisions to go before we can turn it on.

## Link to Issue
Closes:  https://github.com/hicommonwealth/commonwealth/issues/9290

Also fixes issues:

- https://github.com/hicommonwealth/commonwealth/issues/9249
- https://github.com/hicommonwealth/commonwealth/issues/9205
- https://github.com/hicommonwealth/commonwealth/issues/9216

## Description of Changes
- See the v3 issues link above.
- 

## Test Plan

The test plan is in the README.md but here is the list. 

I would not do this yet though.  We can do a full QA and test of everything right before we're ready to turn everything on next week.

### Desktop

- success: copy a .md file to the clipboard, try to paste it into the editor. It
  should insert the content at the editor's cursor

    - this works via a File object (not text) so it's important to test this path.

- success: drag a .md file on top of the editor.  The drag indicator should show
  up and cover the editor while you're dragging.  Then the file should be inserted
  at the cursor.

- success: use the 'Import markdown' button to upload a file.

- success: right click and copy an image in the browser, this should upload it
to the editor and insert it at the current point (I use msnbc.com for this as
their images are copyable and not CSS background images)

- success: take a screenshot, try to paste it into the editor. The upload
  indicator should show up.

- success: use the image button at the top to manually upload an image. The
  upload indicator should show up while this is happening.

- success: drop an image file.  Should upload it for us and not handle it as
  markdown.

- failure: copy multiple .md files ot the clipboard, try to paste into the editor.
  It should fail because we can't handle multiple .md files

### Mobile

It's probably best to test this on a REAL mobile browser (not on a desktop).

- The toolbar should be present at the bottom of the UI.

- They keyboard should stay on top of the keyboard.

## Viewer

- make sure images, tables, and code don't have UI control elements

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Comment by masvelio** [comment]
@burtonator Mobile issues tested on iphone:
1. Bottom toolbar is not visible when i want to start typing and the keyboard shows up
2. I am not able to set the cursor with my finger and select the range of text

https://github.com/user-attachments/assets/f3e2239c-5170-408f-9d9d-e4c11b2de87c

**Comment by burtonator** [comment]
@masvelio The browser you used is mobile Safari right ?

I don't get those issues on Chrome but I do get some different issues.

I'll pull out my tablet/iphone and test there too but I'm going to do it in v4.

Could you approve this PR though so we can push forward?

**Comment by masvelio** [comment]
@burtonator 

> @masvelio The browser you used is mobile Safari right ?

this was on iphone/chrome

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/hooks/useDraft.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
-const KEY_VERSION = 'v2'; // update this for breaking changes
-const PREFIX = `cw-draft-${KEY_VERSION}`;
-
 const MAX_DRAFT_SIZE = 1024 * 1024 * 4;
 
-export function useDraft<T>(key: string) {
-  const fullKey = `${PREFIX}-${key}`;
+type DraftOpts = {
+  keyVersion?: string;
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/hooks/useDraft.tsx:4]: can we set this as default?

File: packages/commonwealth/client/scripts/views/pages/QuillPage/QuillPage.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { ReactQuillEditor } from 'views/components/react_quill_editor';
+
+export const QuillPage = () => {
+  return (
+    <ReactQuillEditor
+      contentDelta=""
+      tooltipLabel="thisis the tooltip"
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/QuillPage/QuillPage.tsx:8]: missing space

**Review by mzparacha — APPROVED** [review]
Left some minor suggestions above. LGTM

**Pull request merged**